text/204-golang-bindings.md

```diff
@@ -0,0 +1,762 @@
+struct, i.e.:
+
+```go
+serviceProps := ecs.BaseServiceProps{
+    ServiceName:       "myService",
+    MaxHealthyPercent: 100,
+    MinHealthyPercent: 50,
+    LaunchType: "EC2",
+}
+
+ecs.TakesBaseServicePropsIface(serviceProps)
+```
+
+The second approach is not to embed the struct, but **flatten** all the properties inherited from the extended interface.
+(See: [Go playground example](https://play.golang.org/p/ioL4XRpjETA)).
+
+```go
+type BaseServiceOptions struct {
+    Cluster                ICluster
+    DesiredCount           *int
+    ServiceName            *string
+    MaxHealthyPercent      *int
+    MinHealthyPercent      *int
+    HealthCheckGracePeriod *Duration
+    CloudMapOptions        CloudMapOptions
+    PropagateTags          PropagatedTagSource
+    EnableECSManagedTags   *bool
+    DeploymentController   DeploymentController
+}
+
+type BaseServiceProps struct {
+    // Flattened properties generated from extended interface (i.e. Base ServiceOptions)
+    Cluster                ICluster
+    DesiredCount           *int
+    ServiceName            *string
+    MaxHealthyPercent      *int
+    MinHealthyPercent      *int
+    HealthCheckGracePeriod *Duration
+    CloudMapOptions        CloudMapOptions
+    PropagateTags          PropagatedTagSource
+    EnableECSManagedTags   *bool
+    DeploymentController   DeploymentController
+
+    LaunchType          LaunchType
+}
+
+func (o BaseServiceOptions) GetServiceName() string    { return o.ServiceName }
+func (o BaseServiceOptions) GetMaxHealthyPercent() int { return o.MaxHealthyPercent }
+func (o BaseServiceOptions) GetMinHealthyPercent() int { return o.MinHealthyPercent }
+
+// ... etc
+
+// Generated interface methods inherited from BaseServiceOptionsIface
+func (p BaseServiceProps) GetServiceName() string    { return p.ServiceName }
+func (p BaseServiceProps) GetMaxHealthyPercent() int { return p.MaxHealthyPercent }
+func (p BaseServiceProps) GetMinHealthyPercent() int { return p.MinHealthyPercent }
+func (p BaseServiceProps) GetLaunchType() string     { return p.LaunchType }
+
+// ... etc
+```
+
+This approach would allow for properties to be passed to a function in a flattened data structure, i.e.
+
+```go
+serviceProps := ecs.BaseServiceProps{
+    ServiceName:       "myService",
+    MaxHealthyPercent: 100,
+    MinHealthyPercent: 50,
+    LaunchType: "EC2",
+}
+
+ecs.TakesBaseServicePropsIface(serviceProps)
+```
+
+The disadvantage is that there is much more boilerplate generated to implement the inherited methods, and any change to the inherited interface would
+be a breaking change to anything inheriting it. For the latter concern, within the jsii, any generated code would be tied to some version of jsii, so
+we would be able to re-generate datatype interfaces that extend other datatypes. For customers creating their own custom constructs, they could
+mititage potential breaking changes in jsii interfaces by creating their own interface wrappers, e.g.
+
```

Comment:
Customers can also mitigate this issue by always ensuring the data-type types that they "extend" embed the generated CDK type.  If they are doing this then they don't need to worry about data types having additional parameters added.

I'd suggest this over customer implementing their own interfaces. for small things implementing their own interfaces might happen, but for large data types i doubt users would want to do this.